lightyear/src/client/prediction/rollback.rs

```diff
@@ -932,6 +943,109 @@ mod integration_tests {
         (stepper, confirmed, predicted)
     }
 
+    /// Test that the entities of a predicted component marked as to be
+    /// entity-mapped are mapped when rollbacked.
+    #[test]
+    fn test_rollback_entity_mapping() {
+        #[derive(Component, Serialize, Deserialize, Clone, Copy, PartialEq)]
+        struct ComponentWithEntity(Entity);
+
+        impl MapEntities for ComponentWithEntity {
+            fn map_entities<M: bevy::prelude::EntityMapper>(&mut self, entity_mapper: &mut M) {
+                self.0 = entity_mapper.map_entity(self.0);
+            }
+        }
+
+        let mut stepper = BevyStepper::default();
+        stepper
+            .client_app
+            .register_component::<ComponentWithEntity>(ChannelDirection::Bidirectional)
+            .add_prediction(ComponentSyncMode::Full)
+            .add_map_entities();
+        stepper
+            .server_app
+            .register_component::<ComponentWithEntity>(ChannelDirection::Bidirectional)
+            .add_prediction(ComponentSyncMode::Full)
+            .add_map_entities();
+        let tick = stepper.client_tick();
+        let confirmed = stepper
```

Comment:
Maybe this whole predicted/confirmed setup is easier to do by just replicating a server entity to the client? not sure